app/src/main/java/com/aptoide/uploader/apps/InstalledApp.java

```diff
@@ -1,39 +1,217 @@
 package com.aptoide.uploader.apps;
 
 import android.content.pm.ApplicationInfo;
+import android.content.pm.PackageInfo;
+import android.content.pm.PackageManager;
 import android.os.Build;
+import android.os.Environment;
+import androidx.annotation.NonNull;
 import androidx.annotation.RequiresApi;
+import androidx.room.Entity;
+import androidx.room.Ignore;
+import androidx.room.PrimaryKey;
+import java.io.File;
 import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.List;
+import org.jetbrains.annotations.NotNull;
+
+@Entity(tableName = "installed") public class InstalledApp {
+  @Ignore public static final int STATUS_COMPLETED = 4;
+  @PrimaryKey @NonNull private String packageAndVersionCode;
+  private String packageName;
+  private String name;
+  private String versionName;
+  private int versionCode;
+  private boolean isSystemApp;
+  private String apkPath;
+  private String iconPath;
+  private long installedDate;
+  @Ignore private Obb obbMain;
+  @Ignore private Obb obbPatch;
+  private boolean isUploaded;
+  private int status;
+  @Ignore private ApplicationInfo appInfo;
 
-public class InstalledApp {
+  public InstalledApp() {
+  }
 
-  private final String name;
-  private final boolean isSystem;
-  private final String packageName;
-  private final String apkPath;
-  private final long installedDate;
-  private final int versionCode;
-  private final Obb obbMain;
-  private final Obb obbPatch;
-  private boolean isUploaded;
-  private ApplicationInfo appInfo;
+  public InstalledApp(PackageInfo packageInfo, PackageManager packageManager) {
+    setAppInfo(packageInfo.applicationInfo);
+    setPackageAndVersionCode(packageInfo.packageName + packageInfo.versionCode);
+    setPackageName(packageInfo.packageName);
+    setName(appInfo.loadLabel(packageManager)
+        .toString());
+    setVersionName(packageInfo.versionName);
+    setVersionCode(packageInfo.versionCode);
+    setIsSystemApp((appInfo.flags & appInfo.FLAG_SYSTEM) != 0);
+    setApkPath(appInfo.sourceDir);
+    setIconPath(
+        "android.resource://" + packageInfo.packageName + "/" + packageInfo.applicationInfo.icon);
+    setInstalledDate(packageInfo.lastUpdateTime);
+    setIsUploaded(false);
+    setObbMain(getMainObb(packageName));
+    setObbPatch(getPatchObb(packageName));
+    setStatus(STATUS_COMPLETED);
+  }
```

Comment:
Is this needed?

Reply:
Dont know exactly what you mean, but im guessing that you are asking if the change was needed.
If so, currently the constructor receives every variable in there and the PackageManagerInstalledAppsProvider would take care of this implementation and know, InstalledIntentService also needs this construtor and would have to take care of this implementation the same way PackageManagerInstalledAppsProvider does. 
With this change, we avoid code duplication by only giving the constructor two variables and the InstalledApp takes care of the variable setters for both PackageManagerInstalledAppsProvider and InstalledIntentService.

Reply:
Passing the package manager inside a pojo is a big nono. the InstalledApp should be an incredible simple object, should not have anything from the Android framework in it. 
Also we are aggregatting the DB with pojo's. DB object and pojo's (which are basically info holding objects to be processed in the business logic part) should not be the same.